src/nnet3/nnet-test-utils.cc

```diff
@@ -1026,12 +1026,21 @@ static void GenerateRandomComponentConfig(std::string *component_type,
     case 24: {
       *component_type = "RepeatedAffineComponent";
       int32 num_repeats = RandInt(1, 50),
-          input_dim = num_repeats * RandInt(1, 15),
-          output_dim = num_repeats * RandInt(1, 15);
```

Comment:
I'll look into the details of what exactly triggers the failure and get
some numbers soon.

On Thu, Dec 31, 2015 at 3:35 PM, Daniel Povey <notifications@github.com>
wrote:

> In src/nnet3/nnet-test-utils.cc
> <https://github.com/kaldi-asr/kaldi/pull/427#discussion_r48670460>:
>
> > @@ -1026,12 +1026,21 @@ static void GenerateRandomComponentConfig(std::string *component_type,
> >      case 24: {
> >        *component_type = "RepeatedAffineComponent";
> >        int32 num_repeats = RandInt(1, 50),
> > -          input_dim = num_repeats * RandInt(1, 15),
> > -          output_dim = num_repeats * RandInt(1, 15);
>
> It prints out the measured and predicted objf changes- how different were
> they? Were they at least usually the same sign? If they were just a bit out
> of the tolerance it's OK to disable that test case, but if they were
> completely wrong, it could indicate a problem e.g. with the derivatives for
> the bias. Dan
> … <#151f9c09c98656f2_>
> On Thu, Dec 31, 2015 at 9:48 AM, Daniel Galvez <notifications@github.com>
> wrote: In src/nnet3/nnet-test-utils.cc <#427 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/427#discussion_r48664776>>: > @@
> -1026,12 +1026,21 @@ static void GenerateRandomComponentConfig(std::string
> *component_type, > case 24: { > *component_type =
> "RepeatedAffineComponent"; > int32 num_repeats = RandInt(1, 50), > -
> input_dim = num_repeats * RandInt(1, 15), > - output_dim = num_repeats *
> RandInt(1, 15); The numerical routines (i.e., Propagate and Backprop) did
> not crash when input_dim=1 and output_dim=1. Rather, the crash in
> ./nnet-component-test was because predicted_obj_change was not
> approximately equal to measured_objf_change in
> TestSimpleComponentDataDerivative(). Obviously, the test works for other
> (larger) input parameters. Simply removing a test case that doesn't work
> seems a little bad, but I think the test is likely to be a false positive
> since the tests work for other tested parameters. Of course I also don't
> really understand the tests as well as I should. (I don't understand why
> the output_deriv-output dot product is a valid objective function.) — Reply
> to this email directly or view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/427/files#r48664776>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/427/files#r48670460>.
>



-- 
Daniel Galvez
danieltgalvez.com